src/main/scala/edu/berkeley/ce/sparkrocks/SeedJointSelector.scala

```diff
@@ -3,242 +3,221 @@ package edu.berkeley.ce.sparkrocks
+    * joints that divide input volume into approximately equal volumes.
     *
-    * @param jointSet Seq of Joints representing a persistent joint set
-    * @param rockVolume Block representing input rock volume that should be divided into approximately equal pieces
-    *                   for each process to operate on
-    * @param numProcessors Number of processes that will be used in analysis
-    * @param tolerance Tolerance in deviation from ideal volume per processor. Default initial value is 1%, though it
-    *                  will be updated as necessary in recursive calls.
-    * @param stepSize Step size that tolerance will be incremented by if unable to find enough seed joints. Default
-    *                 initial value is 1%, though it will be updated as necessary in recursive calls.
-    * @return If able to find the required number of seed joints, this function will return a Seq of Joints. If unable
-    *         to find sufficient seed joints, it will return None.
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numPartitions Number of partitions desired
+    * @param jointSetSpan Number of joint sets to search at a time for partition generation
+    * @param seedBlocks Blocks that will serve as partitions
+    * @param remainingJoints Seq of joints that were not used in seed block generation
+    * @return Returns a tuple containing a Seq of blocks and a Seq of joints. The Seq of blocks
+    *         are the the desired partitions. The Seq of joints are the remaining joints that were
+    *         not used to generate the partitions.
     */
   @tailrec
-  private def findSeedJoints(jointSet: Seq[Joint], rockVolume: Block, numProcessors: Integer, tolerance: Double = 0.01,
-                             stepSize: Double = 0.01): Option[Seq[Joint]] = {
-    val totalVolume = rockVolume.volume
-    val volumePerProc = rockVolume.volume / numProcessors
-    val seedJoints = selectSeedJoints(jointSet, rockVolume, volumePerProc, totalVolume,
-      Seq.empty[Joint], tolerance, numProcessors)
-
-    if (stepSize < 0.01) {
-      // Unable to find satisfactory seed joints using current joint set
-      None
-    } else if (seedJoints.length != numProcessors - 1) {
-      val newStepSize = if (tolerance >= 1.0) { stepSize / 2.0 } else { stepSize }
-      val newTolerance = if (tolerance >= 1.0) { newStepSize } else { tolerance + stepSize }
-      findSeedJoints(jointSet, rockVolume, numProcessors, newTolerance, newStepSize)
-    } else {
-      val initialBlock = Seq(rockVolume)
-      val blocks = seedJoints.foldLeft(initialBlock) { (currentBlocks, joint) =>
+  def findSeedBlocks(jointSets: Seq[Seq[Joint]], inputVolume: Block, numPartitions: Int, jointSetSpan: Int,
+                     seedBlocks: Seq[Block] = Seq.empty[Block], remainingJoints: Seq[Joint] = Seq.empty[Joint]):
+  (Seq[Block], Seq[Joint]) = {
```

Comment:
Done. Was just rolling with what Intellij thought was the most appropriate, in my defense :)